modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -2248,6 +2256,46 @@ private static PartitionSet extendPartitionSet(@Nullable PartitionSet oldPartiti
         return newPartitionSet;
     }
 
+    private void executeIfLocalNodeIsPrimaryForGroup(
```

Comment:
I totally missed that part. I didn't expect that you will update TableManager. Why do you need this?

Reply:
In `TableManager` we do the general logic of rebalance after new pending assignments handling. Before the PR we're looking for RAFT-group leader and then trying to do direct `changePeersAndLearnersAsync` call on leader's client. Now we're looking for primary replica (cited method do this) and then sending the new primary replica request. If somewhy this general approach failed then we're waiting for the failover logic is triggered.

Reply:
Yep, I mean that now it's PrimaryReplica to watch pending changes instead of TableManager. Anyway let it be as you've implemented it for now in order not to inflate the ticket.